### PR TITLE
 ClientID and sessionID accessibility tests implemented

### PR DESCRIPTION
**Pull request by nmarkovic04**

### CODE REVIEW

**Comment by nmarkovic04** [comment]
@rubikzube Rebased.

**Comment by rubikzube** [comment]
I think there's a misunderstanding of what the Client ID is... The Client ID passed to the server in the `i` parameter should be a randomly generated GUID that is not configurable by the customer and is the same for all requests from a given app.

File: AutocompleteClient/FW/Logic/ClientID/ClientIDGenerator.swift [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+//  Copyright © Constructor.io. All rights reserved.
+//  http://constructor.io/
+//
+
+import Foundation
+import UIKit
+
+class ClientIDGenerator: UserIDGenerator{
+    func generateClientID() -> String?{
+        return UIDevice.current.identifierForVendor?.uuidString
```
Comment (rubikzube) [inline AutocompleteClient/FW/Logic/ClientID/ClientIDGenerator.swift]: This `UIDevice.current.identifierForVendor` according to Apple is the same across all apps.  

For constructor.io's purposes we need the client identifier to be different in different apps.  This is a security consideration for our customers.  We do not want to be able to track user behavior across different customers.

For every `device/app` combination that uses constructor, we need a different client ID.  We are also fine if the client ID changes with every install.  In the android client and in the JS client we use random UUID's and store them

File: AutocompleteClient.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1611,10 +1643,11 @@
 		F64F46F41F59CE750094C697 /* Release */ = {
 			isa = XCBuildConfiguration;
 			buildSettings = {
+				ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES = YES;
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
 				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Developer";
 				CODE_SIGN_STYLE = Automatic;
-				DEVELOPMENT_TEAM = "";
+				DEVELOPMENT_TEAM = GK53T2PMMR;
```
Comment (rubikzube) [inline AutocompleteClient.xcodeproj/project.pbxproj]: What are the ramifications of setting DEVELOPMENT_TEAM?

File: AutocompleteClient/FW/Logic/Worker/ConstructorIO.swift [hunk 3]
```diff
@@ -20,12 +20,18 @@ public class ConstructorIO: AbstractConstructorDataSource, CIOTracker, CIOSessio
 
     public static var logger: CIOLogger = CIOPrintLogger()
     
-    private let networkClient: NetworkClient = DependencyContainer.sharedInstance.networkClient()
-    private let sessionManager: SessionManager = DependencyContainer.sharedInstance.sessionManager()
+    private let networkClient: NetworkClient
+    public let sessionManager: SessionManager
```
Comment (rubikzube) [inline AutocompleteClient/FW/Logic/Worker/ConstructorIO.swift]: Does this need to be public?

File: AutocompleteClientTests/Logic/Session/SessionTests.swift [hunk 4]
```diff
@@ -102,4 +102,10 @@ class SessionTests: XCTestCase {
         XCTAssertGreaterThan(nextSession, initialSession, "After timeout is reached, session should be larger than the previous value." )
     }
     
+    func test_SessionIDCanBeRetreivedFromConstructorObject(){
```
Comment (rubikzube) [inline AutocompleteClientTests/Logic/Session/SessionTests.swift]: Not sure what this test does

**Pull request merged**